mitiq/benchmarks/w_state_circuits.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2023 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+""" Functions for creating 2 W-state benchmarking circuits as defined in
+:cite:`Cruz_2019_Efficient`"""
+
+from typing import Optional, Union
+import numpy.typing as npt
+import numpy as np
+import cirq
+
+from mitiq import QPROGRAM
+from mitiq.interface import convert_from_mitiq
+
+
+class GRotationGate(cirq.Gate):
```

Comment:
I think this gate equivalent to the existing Ry gate (https://quantumai.google/reference/python/cirq/Ry), up to a change of variable:
`cos(t/2) <--> p`
If so, it's better to use Ry instead of defining a custom gate.

Reply:
Thanks for catching this ! Defining the custom gate did cause quite a few issues. 

I think, the equivalence you have should be `cos(t/2) <--> SQRT(p)`